File: Sharebook/Server/src/Sharebook.Streams/SampleStream.cs
namespace WorldFeed.Streams
{
    using System.Threading.Tasks;

    using WorldFeed.Common.Helpers;
    using WorldFeed.Common.Public;
    using WorldFeed.Common.Public.Client.Tools;
    using WorldFeed.Common.Public.Parameters.StreamsClient;
    using WorldFeed.Common.Public.Streaming;
    using WorldFeed.Common.Streaming;
    using WorldFeed.Common.Wrappers;
    using WorldFeed.Streams.Properties;

    public class SampleStream : TweetStream, ISampleStream
    {
        public SampleStream(ITwitterClient twitterClient, IStreamResultGenerator streamResultGenerator, IJsonObjectConverter jsonObjectConverter,
            IJObjectStaticWrapper jObjectStaticWrapper, ITwitterClientFactories factories, ICreateSampleStreamParameters createSampleStreamParameters)
            : base(twitterClient, streamResultGenerator, jsonObjectConverter, jObjectStaticWrapper, factories, createSampleStreamParameters)
        {
        }

        public async Task StartAsync()
        {
            await StartAsync(Resources.Stream_Sample).ConfigureAwait(false);
        }
    }
}

File: Sharebook/Client/src/app/streams/SampleStream.ts
// import {ISampleStream} from "../core/Public/Streaming/ISampleStream";
// import {ITwitterClient} from "../core/Public/ITwitterClient";
// import {IStreamResultGenerator} from "../core/Core/Streaming/IStreamResultGenerator";
// import {ITwitterClientFactories} from "../core/Public/Client/Tools/ITwitterClientFactories";
// import {ICreateSampleStreamParameters} from "../core/Public/Parameters/StreamsClient/CreateSampleStreamParameters";
// import {StreamResources} from "./Properties/stream-resources";
// import {TweetStream} from "./TweetStream";
//
// export class SampleStream extends TweetStream implements ISampleStream {
//   constructor(
//     twitterClient: ITwitterClient,
//     streamResultGenerator: IStreamResultGenerator,
//     jsonObjectConverter: IJsonObjectConverter,
//     jObjectStaticWrapper: IJObjectStaticWrapper,
//     factories: ITwitterClientFactories,
//     createSampleStreamParameters: ICreateSampleStreamParameters) {
//     super(twitterClient, streamResultGenerator, jsonObjectConverter,
//       jObjectStaticWrapper, factories, createSampleStreamParameters);
//   }
//
//         public async  StartAsync(): Promise<void>
//         {
//             await super.StartAsync(StreamResources.Stream_Sample).ConfigureAwait(false);
//         }
//     }
